### PR TITLE
Jormun : Add stop_area comments with bragi

### PR DESCRIPTION
**Pull request by benoit-bst**

### CODE REVIEW

File: source/jormungandr/jormungandr/interfaces/v1/serializer/geocode_json.py [hunk 1]
```diff
@@ -217,9 +217,19 @@ class StopAreaSerializer(serpy.DictSerializer):
     timezone = NestedPropertyField(attr='properties.geocoding.timezone')
     commercial_modes = NestedDictGenericField(attr='properties.geocoding.commercial_modes', many=True)
     physical_modes = NestedDictGenericField(attr='properties.geocoding.physical_modes', many=True)
+    comments = NestedDictCommentField(attr='properties.geocoding.comments', many=True)
+    comment = jsonschema.MethodField(display_none=True)
```
Comment (TeXitoi) [inline source/jormungandr/jormungandr/interfaces/v1/serializer/geocode_json.py:221]: in jormungandr: `comment = FirstCommentField(attr='comments', display_none=False, deprecated=True)`

I need a corresponding test.

**Review by TeXitoi — CHANGES_REQUESTED** [review]

File: source/jormungandr/jormungandr/interfaces/v1/serializer/geocode_json.py (hunk 1 above)
Comment (benoit-bst) [reply source/jormungandr/jormungandr/interfaces/v1/serializer/geocode_json.py]: l have tried to use this method but it didn't work.
This is the reason why I have created the "get_comment" method in StopAreaSerializer class.

Comment (benoit-bst) [reply source/jormungandr/jormungandr/interfaces/v1/serializer/geocode_json.py:221]: I have tried to use this method, but it didn't work. This is the reason why I have created "get_comment" method in StopAreaSerializer class.
There is a test : https://github.com/CanalTP/navitia/pull/2378/files#diff-e965ac31a14eb9d613cd1e9053897620R1179

Comment (benoit-bst) [reply source/jormungandr/jormungandr/interfaces/v1/serializer/geocode_json.py]: JIRA avec les explications associés :
https://jira.canaltp.fr/browse/NAVP-810

Comment (TeXitoi) [reply source/jormungandr/jormungandr/interfaces/v1/serializer/geocode_json.py:221]: No, I want to say that, if there is no comment, the comment field must not be present.

**Comment by benoit-bst** [comment]
All explanations for QA in JIRA ticket :
https://jira.canaltp.fr/browse/NAVP-810

File: source/jormungandr/jormungandr/interfaces/v1/serializer/geocode_json.py (hunk 1 above)
Comment (TeXitoi) [reply source/jormungandr/jormungandr/interfaces/v1/serializer/geocode_json.py:221]: `display_none=False`

Comment (benoit-bst) [reply source/jormungandr/jormungandr/interfaces/v1/serializer/geocode_json.py:221]: With this PR, when stop area comments exist we display it, but if there is no comments associated, "comments" and "comment" field no exist.

A example with 2 stops area, the first with comments and the second without :

with_and_without_comment

Comment (benoit-bst) [reply source/jormungandr/jormungandr/interfaces/v1/serializer/geocode_json.py:221]: I can add a test when a stop area has no comments.

Comment (benoit-bst) [reply source/jormungandr/jormungandr/interfaces/v1/serializer/geocode_json.py:221]: Added test :white_check_mark:

Comment (TeXitoi) [reply source/jormungandr/jormungandr/interfaces/v1/serializer/geocode_json.py:221]: :+1:

File: source/jormungandr/tests/bragi_autocomplete_tests.py [hunk 2]
```diff
@@ -559,6 +559,40 @@ def geojson():
...
+                "coordinates": [
+                    3.282103,
+                    49.847586
+                ],
+                "type": "Point"
+            },
+            "properties": {
+                "geocoding": {
+                    "city": "Bobtown",
+                    "housenumber": "20",
```
Comment (TeXitoi) [inline source/jormungandr/tests/bragi_autocomplete_tests.py]: this field does not exists with stop area

File: source/jormungandr/tests/bragi_autocomplete_tests.py [hunk 3]
```diff
@@ -559,6 +559,40 @@ def geojson():
...
+                    "id": "stop_area_id",
+                    "label": "20 Rue Bob (Bobtown)",
+                    "name": "Rue Bob",
+                    "postcode": "02100",
+                    "street": "Rue Bob",
```
Comment (TeXitoi) [inline source/jormungandr/tests/bragi_autocomplete_tests.py]: idem

**Review by TeXitoi — COMMENTED** [review]
Just check that the mocks are realistic, and OK

File: source/jormungandr/tests/bragi_autocomplete_tests.py (hunk 2 above)
Comment (benoit-bst) [reply source/jormungandr/tests/bragi_autocomplete_tests.py]: I have cleaned the mock to be realistic
:white_check_mark:

File: source/jormungandr/tests/bragi_autocomplete_tests.py (hunk 3 above)
Comment (benoit-bst) [reply source/jormungandr/tests/bragi_autocomplete_tests.py]: :white_check_mark:

**Review by TeXitoi — APPROVED** [review]

File: source/jormungandr/tests/bragi_autocomplete_tests.py [hunk 4]
```diff
@@ -1134,6 +1268,64 @@ def test_autocomplete_for_admin_depth_two(self):
...
+            is_valid_global_autocomplete(response, depth=1)
+            r = response.get('places')
+            assert len(r) == 1
+            assert r[0]['name'] == 'stop 1'
+            assert r[0]['embedded_type'] == 'stop_area'
+            assert r[0]['id'] == 'stop_area_id'
+            assert r[0]['quality'] == 0 # field for kraken compatibility (default = 0)
+            stop_area = r[0]['stop_area']
+            # When no comments exist in Bragi, the API mask the "comment" and "comments" field.
+            assert not stop_area.get('comment')
```
Comment (pbougue) [inline source/jormungandr/tests/bragi_autocomplete_tests.py:1326]: the test should probably be `assert 'comment' not in stop_area` (same below)

File: source/jormungandr/jormungandr/interfaces/v1/serializer/geocode_json.py (hunk 1 above)
Comment (pbougue) [reply source/jormungandr/jormungandr/interfaces/v1/serializer/geocode_json.py:221]: Did you try without the `display_none` ? It's probably useless, if it works the way we want, so we might as well remove it.

**Pull request merged**